examples/lighting-app/silabs/src/LightingManager.cpp

```diff
@@ -307,3 +359,56 @@ void LightingManager::OnTriggerOffWithEffect(OnOffEffect * effect)
     LightMgr().mOffEffectArmed = true;
     LightMgr().StartTimer(offEffectDuration);
 }
+
+#if (defined(SL_MATTER_RGB_LED_ENABLED) && SL_MATTER_RGB_LED_ENABLED == 1)
+bool LightingManager::InitiateLightctrlAction(int32_t aActor, Action_t aAction, uint32_t aAttributeId, uint8_t * value)
+{
+    bool action_initiated = false;
+    VerifyOrReturnError(aAction == COLOR_ACTION_XY || aAction == COLOR_ACTION_HSV || aAction == COLOR_ACTION_CT, action_initiated);
+    RGBLEDWidget::ColorData_t colorData;
+    switch (aAction)
+    {
+    case COLOR_ACTION_XY:
+        colorData.xy = {};
+
+        if (aAttributeId == ColorControl::Attributes::CurrentX::Id)
+        {
+            VerifyOrReturnValue(mCurrentX != *reinterpret_cast<uint16_t *>(value), action_initiated);
+            mCurrentX = *reinterpret_cast<uint16_t *>(value);
```

Comment:
You should just set the object value when they are applied to the LED, so in the `PostLightControlActionRequest`.
So here and in the other cases I would only set the temporary value passed as event.
for example here I would init colorData.xy with the object variable and then update colorData.xy based on the received attribute change.

```
colorData.xy = { mCurrentX, mCurrentY };
if (aAttributeId == ColorControl::Attributes::CurrentX::Id)
{
    ...
    colorData.xy.x = *reinterpret_cast<uint16_t *>(value);
}
...
```
Else you can get odd behaviour where to mCurrentX or mCurrentY are changed while you are applying it

Reply:
Yeah changed for all colorData variables.